src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/config/WebClientConfiguration.kt

```diff
@@ -14,37 +15,49 @@ import org.springframework.web.reactive.function.client.ExchangeStrategies
 import org.springframework.web.reactive.function.client.WebClient
 
 @Configuration
-class WebClientConfiguration(@Value("\${api.base.url.nomis}") val baseUri: String) {
+class WebClientConfiguration(
+  @Value("\${api.base.url.nomis}") val prisonBaseUri: String
+) {
 
   @Bean
-  fun prisonWebClient(authorizedClientManager: OAuth2AuthorizedClientManager?): WebClient? {
-    val oauth2Client = ServletOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager)
-    oauth2Client.setDefaultClientRegistrationId("nomis-api")
+  fun prisonWebClient(authorizedClientManager: OAuth2AuthorizedClientManager?): WebClient? =
+    buildWebClient(prisonBaseUri, authorizedClientManager)
 
-    val exchangeStrategies = ExchangeStrategies.builder()
-      .codecs { configurer: ClientCodecConfigurer -> configurer.defaultCodecs().maxInMemorySize(-1) }
-      .build()
-
-    return WebClient.builder()
-      .baseUrl(baseUri)
-      .apply(oauth2Client.oauth2Configuration())
-      .exchangeStrategies(exchangeStrategies)
-      .build()
-  }
+  @Bean
+  @ConditionalOnProperty(value = ["api.base.url.restricted-patients"], havingValue = "true")
```

Comment:
The the api is only available in dev at the moment. Once pre and prod have been configured I'll pull the value up into the constructor.

Reply:
I think `havingValue` does a match on the property value i.e. in this case the value of the property must be `true`.  If you remove `havingValue` then it just checks if the property exists.